src/scripts/main.js

```diff
@@ -1,3 +1,262 @@
+    return;
+  }
+
+  const boardClone = JSON.stringify(board);
+
+  switch (e.key) {
+    case 'ArrowLeft':
+      slideLeft();
+      break;
+
+    case 'ArrowRight':
+      slideRight();
+      break;
+
+    case 'ArrowUp':
+      slideUp();
+      break;
+
+    case 'ArrowDown':
+      slideDown();
+      break;
+  }
+
+  if (boardClone === JSON.stringify(board)) {
+    return;
+  }
+
+  addNumber();
+
+  if ([...cells].some(cell => cell.classList.contains('field-cell--2048'))) {
+    winMessage.classList.remove('hidden');
+    gameEnded = true;
+  }
+
+  if (!moveIsPossible()) {
+    loseMessage.classList.remove('hidden');
+  }
+});
+
+function clearZeroes(row) {
+  return row.filter(element => element !== 0);
+}
+
+function slide(row) {
+  let newRow = clearZeroes(row);
+
+  for (let i = 0; i < newRow.length; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow[i + 1] = 0;
+
+      tableScore.textContent = +tableScore.textContent + newRow[i];
+    }
+  }
+
+  newRow = clearZeroes(newRow);
+
+  while (newRow.length < 4) {
+    newRow.push(0);
+  }
+
+  return newRow;
+}
+
+function slideLeft() {
+  for (let row = 0; row < 4; row++) {
+    const fieldRow = slide(board[row]);
+
+    board[row] = fieldRow;
+
+    for (let column = 0; column < 4; column++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
+}
+
+function slideRight() {
+  for (let row = 0; row < 4; row++) {
+    const fieldRow = slide(board[row].reverse());
+
+    board[row] = fieldRow.reverse();
+
+    for (let column = 0; column < 4; column++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
+}
+
+function slideUp() {
+  for (let column = 0; column < 4; column++) {
+    let fieldColumn = [board[0][column], board[1][column],
+      board[2][column], board[3][column]];
+
+    fieldColumn = slide(fieldColumn);
+
+    board[0][column] = fieldColumn[0];
+    board[1][column] = fieldColumn[1];
+    board[2][column] = fieldColumn[2];
+    board[3][column] = fieldColumn[3];
+
+    for (let row = 0; row < 4; row++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
+}
+
+function slideDown() {
+  for (let column = 0; column < 4; column++) {
+    let fieldColumn = [board[0][column], board[1][column],
+      board[2][column], board[3][column]];
+
+    fieldColumn = slide(fieldColumn.reverse()).reverse();
+
+    board[0][column] = fieldColumn[0];
+    board[1][column] = fieldColumn[1];
+    board[2][column] = fieldColumn[2];
+    board[3][column] = fieldColumn[3];
+
+    for (let row = 0; row < 4; row++) {
+      const tile = rows[row].children[column];
+      const number = board[row][column];
+
+      updateTile(tile, number);
+    }
+  }
```

Comment:
You repeat the same logic for the slideUp(). Can you create a separate method for this logic?